text/0071-continue-trace-over-process-boundaries.md

```diff
@@ -0,0 +1,93 @@
+
+When `SENTRY_TRACING_USE_ENVIRONMENT` is set to `true` during startup also a so called "Root Transaction" should be created automatically. It should include all the trace information from the parsed tracing information and should be finished just before the process ends.
+
+TODO: Maybe we should create a new `transaction_info.source` for this kind of root transactions?
+
+### Creating tracing information on process start up
+
+If `SENTRY_TRACING_USE_ENVIRONMENT` is set to `true` and no information can be found in `SENTRY_TRACING_BAGGAGE` or `SENTRY_TRACING_SENTRY_TRACE` then the current process is the head of trace and a dynamic sampling context should be created.
+
+See [Unified Propagation Mechanism](https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#unified-propagation-mechanism) for details.
+
+See [Dynamic Sampling Context Payloads](https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#payloads) to see what data needs to be included in the dynamic sampling context.
+
+### Propagating/sending tracing information via environment variables:
+
+The integrations that patch functions that are used for spawning new processes (`StdlibIntegration` in Python) should be changed so they use the parsed tracing information (if any) and serialize it to the environment variables (`SENTRY_TRACING_BAGGAGE`, `SENTRY_TRACING_SENTRY_TRACE`) for the newly spawned process. The variable `SENTRY_TRACING_USE_ENVIRONMENT` should also be set to `true` so the receiving process is picking up the information.
+
+# Drawbacks
+
+Because the serialization format of the bagagge (and thus the dynamic sampling context) stays the same, just the carrier is a new one, it should be 100% compatible to other integrations and will not break dynamic sampling.
+
+So: No drawbacks.
+
+# Unresolved questions
+
+- Should we create a new `transaction_info.source` for this kind of transactions that span one execution of a process?
+
+# Appendix A
+
+How trace propagation is working in the Python SDK right now:
+
+When a request from another service is received a transaction is created. (So each request-response cycle is a transaction).
+The process of creating a transaction that attaches to an existing trace received from the calling service is done as follows:
+
+- data from the `baggage` header is parsed. If it contains values with a `sentry-` prefix the parsed baggage is frozen (marked as immutable).
+- data from the `sentry-trace` header is parsed. (`trace_id`, `parent_span_id` and `parent_sampled`) If this header exists and contains values the parsed baggage from step is is frozen (marked as immutable).
+- DEPRECATED and will be removed soon: data from the `tracestate` header is parsed. (`sentry_tracestate` and `third_party_tracestate`).
+- With all this information a new transaction is created.
```

Comment:
Can we have a recommendation of what we call this transaction?

Reply:
this is just a description on how existing integrations work. So there is not a generic term for this transaction because for every SDK it is something different, a request-response-cycle on the server, a user-interaction in the browser and something else on mobile.